### ISSUE TITLE
Make reversing the plot easier?

### ISSUE THREAD

**Issue by andrewheiss** [malcolmbarrett/ggdag#7]
With some DAGs it's helpful to plot them horizontally rather than vertically (e.g. saving space in journals and reports). Flipping the plots is easy enough with `coord_flip()`, but there are a couple problems with doing that.

Here's a reprex:

``` r
library(tidyverse)
library(ggdag)

example <- dagify(Y ~ X,
                  X ~ A + B) %>% 
  tidy_dagitty(layout = "sugiyama")

# Normal top-down graph
ggdag(example)
```

![](https://i.imgur.com/Sc8m62P.png)

Flipping this is as easy as adding `coord_flip()`:

``` r
# Flipped x and y axis so that the DAG is horizontal
ggdag(example) +
  coord_flip()
```

![](https://i.imgur.com/1onAYcC.png)

However, this now reads from right to left, rather than left to right. It would be nice to have A and B on the left and Y on the right since English reads left to right, top to bottom. 

The only way to do this now is to add `scale_y_reverse()` to flip the y axis (which is now horizontal thanks to `coord_flip()`. Doing this creates a message, though, since `scale_y_continuous()` is already set as part of `scale_dag()`:

    Scale for 'y' is already present. Adding another scale for 'y', which will replace the existing scale.

It's easy enough to live with the message and not worry, but `scale_y_reverse()` also needs additional arguments to make it equivalent to just using `scale_dag()`. Without `expand = expand_scale(c(0.1, 0.1))`, the nodes appear too close to the edge of the plot. Adding the `expand` argument works fine and the DAG flips without issues (except for the message):

``` r
ggdag(example) +
  coord_flip() + 
  scale_y_reverse(expand = expand_scale(c(0.1, 0.1)))
#> Scale for 'y' is already present. Adding another scale for 'y', which
#> will replace the existing scale.
```

![](https://i.imgur.com/TltJ7Dg.png)

It would be cool if there were a more systematic way of doing this, like `scale_dag(flip = TRUE)` or something. I don't know the best approach, though, and I don't know if an addition like that fits with the overall philosophy of the package. Plus in some situations, not reversing the y axis could be helpful (Arabic/Hebrew DAGs that should read right to left horizontally). 

<sup>Created on 2019-05-21 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

**Comment by malcolmbarrett** [comment]
hmm, I think I need to think about this. In the dev version, you won't get that message (in part because `scale_dag()` is being deprecated), but you would still need `coord_flip()` + `scale_y_reverse()`

My general philosophy with ggdag is to not overload the fast plotting functions with arguments. I also try not to get in the way of ggplot2's excellent flexibility, so I'm not sure if I want to implement something for which a solution already exists. I'm definitely alright with convenience functions specifically for DAGs, though...

If I did, it might be something simple like

```r
flip_dag <- function(expand = expand_scale(c(0.1, 0.1))) {
  list(
    coord_flip(),
    scale_y_reverse(expand = expand)
  )
}

ggdag(example) + flip_dag()
```

**Comment by andrewheiss** [comment]
Or maybe there's something igraph-esque that will flip/reverse things up in `tidy_daggity()` so there's no need for ggplot adjustments (see https://rdrr.io/cran/DiagrammeR/man/rev_edge_dir.html or https://rdrr.io/rforge/dnet/man/dDAGreverse.html for instance)

**Comment by malcolmbarrett** [comment]
I wondered about that too. The other thing is that these positions are algorithmic, and what reversing will do in some situations is less clear than others. If there's an inherent order in your DAG, maybe you should just be specifying the coordinates directly instead of trying to solve the problem programmatically?

**Comment by malcolmbarrett** [comment]
This second part is related to something else I've thought about: can I specify some positions or an entire axis while letting an algorithm pick the rest (e.g. always have -> left to right but allow algorithmic vertical placements)?

**Comment by andrewheiss** [comment]
Ooh, yeah, similar to `geom_text_repel(direction = "y")`.

Having a horizontal DAG order seems common enough to have this be an option (see @eleanormurray's cool tikz/tex DAGs at https://github.com/eleanormurray/causalgraphs_latex/blob/master/graphical-models-causal.2.7.19.pdf ; but she specifies the coordinates more or less by hand rather than with igraph-y algorithms) ¯\\\_(ツ)\_/¯

**Comment by malcolmbarrett** [comment]
yeah, I have version of her code in [malcolmbarrett/dagtex](github.com/malcolmbarrett/dagtex) that does that by default (but not randomly for `y`). I've thought about how to handle that in ggdag, and I'm just not sure as to the best way.

**Comment by malcolmbarrett** [comment]
Way later, I'm going to close this, because I don't think it should be ggdag's job to guess which orientation you want, particularly for coordinates generated by an algorithm. I think you should explicitly use the ggplot2 functions above. Thanks for filing long ago!
